src/autopas/containers/linkedCells/traversals/C04SoACellHandler.h

```diff
@@ -206,57 +214,54 @@ inline void C04SoACellHandler<ParticleCell, PairwiseFunctor, DataLayout, useNewt
         }
         // two cases intervall in current  stripe
         cell1 = &combinationSlice[currentSlice];
-        cell1->_particleSoABuffer.setViewStart(0);
+        auto stripeView = cell1->_particleSoABuffer.constructView(0, numParticlesBaseCell);
+        cell1ViewStart = 0;
         if (slice == currentSlice) {
           // process stripe with itself
-          cell1->_particleSoABuffer.setViewLength(numParticlesBaseCell);
-
-          this->_cellFunctor.processCell(*cell1);
-
-          cell1->_particleSoABuffer.setViewStart(numParticlesBaseCell);
-          cell1->_particleSoABuffer.setViewLength(-1);
+          _pairwiseFunctor->SoAFunctor(stripeView, useNewton3);
 
-          ParticleCell temp;
-          temp._particleSoABuffer.append(cell1->_particleSoABuffer);
-
-          cell1->_particleSoABuffer.setViewStart(0);
-          cell1->_particleSoABuffer.resizeArrays(numParticlesBaseCell);
-          this->_cellFunctor.processCellPair(*cell1, temp);
-
-          cell1->_particleSoABuffer.append(temp._particleSoABuffer);
+          auto restView = cell1->_particleSoABuffer.constructView(numParticlesBaseCell,
+                                                                  cell1->_particleSoABuffer.getNumParticles());
+          _pairwiseFunctor->SoAFunctor(stripeView, restView, useNewton3);
+          if (not useNewton3) {
```

Comment:
@Garfield96  looking at this traversal in this way it seems Newton3 off does not make too much sense, does it?